src/bp-xprofile/classes/class-bp-xprofile-group.php

```diff
@@ -486,7 +492,12 @@ public static function get_group_ids( $args = array() ) {
 		$bp = buddypress();
 
 		if ( ! empty( $r['profile_group_id'] ) ) {
-			$where_sql = $wpdb->prepare( 'WHERE g.id = %d', $r['profile_group_id'] );
+			if ( is_integer( $r['profile_group_id'] ) ) {
+				$where_sql = $wpdb->prepare( 'WHERE g.id = %d', $r['profile_group_id'] );
+			} else {
+				$profile_group_ids = join( ',', wp_parse_id_list( $r['profile_group_id'] ) );
+				$where_sql         = "WHERE g.id IN ({$profile_group_ids})";
+			}
```

Comment:
Good point! I would be surprised of any performance implications. I think maybe if someone adds hundreds or thousands of ids: https://stackoverflow.com/questions/4514697/mysql-in-operator-performance-on-large-number-of-values

I'll update as suggested since this simplification is a good one.

Reply:
I added a minor change since the boolean `true` could fallback to 1.